echopype/mask/api.py

```diff
@@ -347,44 +354,25 @@ def apply_mask(
     if final_mask_chan_shape != source_da_chan_shape:
         raise ValueError(
             f"The final constructed mask is not of the same shape as source_ds[{var_name}] "
-            "along the ping_time and range_sample dimensions!"
+            "along the ping_time, and range_sample dimensions!"
         )
-
-    # Apply the mask to var_name
-    if "channel" in final_mask.dims and "channel" in source_da.dims:
-        # Identify common channels
-        common_channels = set(final_mask.coords["channel"].values).intersection(
-            set(source_da.coords["channel"].values)
+    # If final_mask has dim channel then source_da must have dim channel
+    if "channel" in final_mask.dims and "channel" not in source_da.dims:
+        raise ValueError(
+            "The final constructed mask has dim channel, "
+            f"so source_ds[{var_name}] must have dim channel."
```

Comment:
Good to write error messages with clear text so users can have clear guidance. Things that are understandable to us may not be necessarily understandable widely.
```suggestion
            "The final constructed mask has the channel dimension, "
            f"so source_ds[{var_name}] must also have the channel dimension."
```